api/src/models/filter_models.rs

```diff
@@ -1,7 +1,48 @@
+use utoipa::openapi::RefOr;
+use utoipa::openapi::schema::{ObjectBuilder, Schema};
+use utoipa::{PartialSchema, ToSchema};
+
 #[derive(serde::Deserialize, Clone)]
 pub struct FilterSubmissionBody {
     pub pipeline: Option<Vec<mongodb::bson::Document>>,
     pub permissions: Option<Vec<i32>>,
     pub catalog: Option<String>,
     pub id: Option<i32>,
 }
+impl ToSchema for FilterSubmissionBody {
```

Comment:
This is quite verbose for converting something that is very similar to a JSON Value into a schema. As discussed with @Theodlz, maybe our functions should accept a simpler type, and we convert it to a MongoDB document internally.